scala_proto/private/scalapb_aspect.bzl

```diff
@@ -0,0 +1,235 @@
+
+        if toolchain.with_flat_package:
+            flags.append("flat_package")
+
+        if toolchain.with_single_line_to_string:
+            flags.append("single_line_to_proto_string")
+
+        # This feels rather hacky and odd, but we can't compare the labels to ignore a target easily
+        # since the @ or // forms seem to not have good equality :( , so we aim to make them absolute
+        #
+        # the backlisted protos in the tool chain get made into to absolute paths
+        # so we make the local target we are looking at absolute too
+        target_absolute_label = target.label
+        if not str(target_absolute_label)[0] == "@":
+            target_absolute_label = Label("@%s//%s:%s" % (ctx.workspace_name, target.label.package, target.label.name))
+
+        for lbl in toolchain.blacklisted_protos:
+            if(lbl.label == target_absolute_label):
+                compile_protos = False
+
+        code_generator = toolchain.code_generator
+        for lbl in toolchain.override_code_generator_targets:
+            if(lbl.label == target_absolute_label):
+                code_generator = toolchain.override_code_generator
+
+
+        for lbl in toolchain.override_code_generator_targets:
+            if(lbl.label == target_absolute_label):
+                code_generator = toolchain.override_code_generator
+
+        if compile_protos:
+            scalapb_file = ctx.actions.declare_file(
+                target.label.name + "_scalapb.srcjar",
+            )
+            proto_to_scala_src(
+                ctx,
+                target.label,
+                code_generator,
+                compile_protos,
+                transitive_protos,
+                target_ti.transitive_proto_path.to_list(),
+                flags,
+                scalapb_file,
+            )
+
+            src_jars = depset([scalapb_file])
+            output = _compiled_jar_file(ctx.actions, scalapb_file)
+            outs = depset([output])
+            java_info = _compile_scala(
+                ctx,
+                toolchain.scalac,
+                target.label,
+                output,
+                scalapb_file,
+                deps,
+                imps,
+            )
+        else:
+            # this target is only an aggregation target
+            src_jars = depset()
+            outs = depset()
+            java_info = _empty_java_info(deps, imps)
+
+    return [
+        ScalaPBAspectInfo(
+            src_jars = src_jars,
+            output_files = outs,
+            proto_info = transitive_ti,
+            java_info = java_info,
+        ),
+    ]
+
+scalapb_aspect = aspect(
+    implementation = _scalapb_aspect_impl,
+    attr_aspects = ["deps"],
+    required_aspect_providers = [
+        [ProtoInfo],
+        [ScalaPBImport],
+    ],
+    attrs = {
+        "_protoc": attr.label(executable = True, cfg = "host", default = "@com_google_protobuf//:protoc"),
```

Comment:
@simuons this is the same change you have, do you agree?

Reply:
Yes it is.